src/test/exporter/test_exporter.cc

```diff
@@ -1452,3 +1479,84 @@ TEST(Exporter, add_fixed_name_metrics) {
     EXPECT_EQ(new_metric.first, expected_labels);
     ASSERT_TRUE(new_metric.second == expected_metric_name);
 }
+
+TEST(Exporter, UpdateSockets) {
+    const std::string mock_dir = "/tmp/fake_sock_dir";
+
+    // Create the mock directory
+    std::filesystem::create_directories(mock_dir);
+
+    // Create a mix of vstart and real cluster mock .asok files
+    std::ofstream(mock_dir + "/ceph-osd.0.asok").close();
+    std::ofstream(mock_dir + "/ceph-mds.a.asok").close();
+    std::ofstream(mock_dir + "/ceph-mds.a.asok").close(); //intentionally repeated
```

Comment:
Just to understand, what's the purpose of this repetition here? could you provide more info in the comment plz?

Reply:
You are right... These are "mock" files in the socket folder. it has no sense at all any repeat...